### PR TITLE
[ci]: Support azp code coverage

### PR DESCRIPTION
**Pull request by xumia**
Support azp code coverage

1. Support full coverage in Azp
2. Support diff coverage in Azp and the check in GitHub

### CODE REVIEW

**Comment by qiluo-msft** [comment]
Is it possible to see the number and code lines of "full coverage in Azp" in this PR's check result page, I cannot find them in https://dev.azure.com/mssonic/build/_build/results?buildId=38907&view=results

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,7 @@ MGMT_COMMON_DIR := $(TOP_DIR)/sonic-mgmt-common
 BUILD_DIR := build/bin
 export CVL_SCHEMA_PATH := $(MGMT_COMMON_DIR)/build/cvl/schema
 export GOBIN := $(abspath $(BUILD_DIR))
+export PATH := $(PATH):$(GOBIN):$(shell dirname $(GO))
```
Comment (qiluo-msft) [inline Makefile:14]: why we need this line?

**Review by lguohan — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (xumia) [reply Makefile:14]: It is to resolve gocov not found issue, for go bin and go command not in the PATH.
gocov is in $(GOBIN), and gocov need the go program which is in $(shell dirname $(GO))

**Comment by xumia** [comment]
> Is it possible to see the number and code lines of "full coverage in Azp" in this PR's check result page, I cannot find them in https://dev.azure.com/mssonic/build/_build/results?buildId=38907&view=results

Yes, it is in the in the coverage report artifact. The html report only shows the DIFF report, but the coverage file is still uploaded to the artifacts.

https://dev.azure.com/mssonic/build/_build/results?buildId=38907&view=artifacts&pathAsName=false&type=publishedArtifacts
coverage38907/coverage.xml

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**